cli/cli/cmds_volume.go

```diff
@@ -339,7 +344,11 @@ func (c *CLI) initVolumeCmds() {
 			// volumes that are attached or can be attached should be valid
 			// candidates for a mount operation
 			result, err := c.lsVolumes(
-				args, volumeStatusAttached, volumeStatusAvailable)
+				args,
+				requestAttachments(
+					apitypes.VolumeAttachmentsMine,
+					apitypes.VolumeAttachmentsAttached,
+					apitypes.VolumeAttachmentsUnattached))
```

Comment:
@akutz 

>I'm fully willing to admit I'm off base on all of this, but I also think it's possible I just did a supremely shitty job describing the purpose and implementation of these things with which to begin, and it's lead to much confusion.

No no, I totally understand the purpose, and think the implementation is great. I just narrowed in on a usecase where I thought the implementation would return invalid results. After going through all your comments, I *think* I've identified the crux of it all.

We have different interpretations of mask `1|2|8`.  You defined this as:
>Request attachment information for all retrieved volumes and return only volumes that attached to the instance provided in the instance ID.

I would have defined it as:
Request attachment information for all retrieved volumes and return volumes that are attached to the instance provided in the instance ID **or attached to a different instance**.

I think `1|2|8` is functionally equivalent to `1|8`, and you think it is equivalent to `1|2`.  From your table:

| Mask | Compnents | Description |
| ---- | ------------ | ----------- |
| `3` | `1|2` | Request attachment information for volumes attached to the instance provided in the instance ID |
| `11` | `1|2|8` | Request attachment information for all retrieved volumes and return only volumes that attached to the instance provided in the instance ID. |

I do not see a difference between those two. Do you? Is there a scenario where those two masks would return different results?

This is why I was saying `Mine` has precedence over `Attached`. In my scenario, it was going to keep volumes attached to other instances from being returned -- I thought that was wrong. You are saying that is by design.

This is totally fine! I just think we need to make that clear. My confusion stemmed from not seeing why you want to use mask `1|2|8` unless you *wanted* to get volumes attached to other instances, because otherwise you would just use mask `1|2`.

Does that make sense?